scripts/cleanup_versions.py

```diff
@@ -0,0 +1,117 @@
+import sys
+import boto3
+
+
+class CleanupVersions:
+    def __init__(self):
+        self.lambda_client = boto3.client("lambda")
+        self.appconfig_client = boto3.client("appconfig")
+        self.environment = sys.argv[1]
+
+    def start(self):
+        self.delete_hosted_configuration_versions()
+        self.delete_lambda_layer_versions()
+
+    def get_app_config_application_id(self) -> str:
+        current_applications = self.appconfig_client.list_applications()
+        for app in current_applications["Items"]:
+            if f"-{self.environment}" in app["Name"]:
+                return app["Id"]
+
+    def get_app_config_profile_id(self, application_id: str) -> str:
+        config_profiles = self.appconfig_client.list_configuration_profiles(
+            ApplicationId=application_id
+        )
+        for profile in config_profiles["Items"]:
+            if f"-{self.environment}" in profile["Name"]:
+                return profile["Id"]
+
+    def get_hosted_configuration_versions(self):
+        print(f"\nGathering AppConfig hosted configuration versions on {self.environment}...")
+        application_id = self.get_app_config_application_id()
+        config_profile_id = self.get_app_config_profile_id(application_id)
+
+        current_hosted_configuration_versions = self.appconfig_client.list_hosted_configuration_versions(
+            ApplicationId=application_id,
+            ConfigurationProfileId=config_profile_id
+        )
+
+        return current_hosted_configuration_versions["Items"]
+
+    def delete_hosted_configuration_versions(self):
+        excess_hosted_config_versions = self.get_hosted_configuration_versions()
+        total_untracked_versions = len(excess_hosted_config_versions)
+        print(f"\n{total_untracked_versions} hosted configuration versions require deletion")
+
+        if not total_untracked_versions:
+            return
+
+        successful_deletes = []
+        print("\nDeleting configuration versions...")
+        for version in excess_hosted_config_versions:
+            response = self.appconfig_client.delete_hosted_configuration_version(
+                ApplicationId=version['ApplicationId'],
+                ConfigurationProfileId=version['ConfigurationProfileId'],
+                VersionNumber=version['VersionNumber']
+            )
+            if response["ResponseMetadata"]["HTTPStatusCode"] == 204:
+                successful_deletes.append(response)
+
+        if total_untracked_versions:
```

Comment:
Do we need this if? we are checking in line 46 `if not total_untracked_versions`

Reply:
No this can be removed, cheers